### ISSUE TITLE
Add `toml!` macro

### ISSUE THREAD

**Issue by killercup** [alexcrichton/toml-rs#195]
Let's shamelessly copy the `json!` macro from serde_json, but for (a subset of) TOML.

**Comment by alexcrichton** [comment]
👍

**Comment by dtolnay** [comment]
TOML syntax inside, right?

```rust
let t = toml! {
  [dependencies]
  toml = "0.4"
};
```

**Comment by killercup** [comment]
@dtolnay if you can manage that, sure! (Proc macros can't come soon enough!)

Until then

> (a subset of) TOML

would still make me happy :)

```rust
// almost json
let t = toml! {
    dependencies = [
        { toml = "0.4" },
    ]
};
```

**Comment by dtolnay** [comment]
I implemented a basic version of this in #217 (with real TOML syntax!). There may be some design work left around how we want to handle interpolations.

```rust
let v = "1.0";

let cargo_toml = toml! {
    [dependencies]
    serde = /* i want to insert the serde version here */
};
```
